packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/AnnotationEditorUtils.ts

```diff
@@ -1,9 +1,66 @@
+import { AdditionalPropertiesSchemaField } from '@/components/SchemaDrivenAnnotationEditor/field/AdditionalPropertiesSchemaField'
 import { entityJsonKeys } from '@/utils/functions/EntityTypeUtils'
 import { RJSFValidationError } from '@rjsf/utils'
 import { ENTITY_CONCRETE_TYPE } from '@sage-bionetworks/synapse-types'
 import { JSONSchema7 } from 'json-schema'
+import jsonpath from 'jsonpath'
 import { flatMap, groupBy, isEmpty } from 'lodash-es'
 
+/**
+ * Generates a JSON schema for the form UI based on the provided validation schema and entity schema base properties.
+ *
+ * To meet the requirements of the desired form to display, we cannot simply show the validation schema. This is because
+ * validation schemas may use conditional logic based on default entity properties (e.g. concreteType) to show/hide fields.
+ * This method combines the validation schema with the entity schema base properties to create a new schema that will yield
+ * the expected form with desired logic.
+ * @param validationSchema
+ * @param entitySchemaBaseProperties
+ */
+export function getJsonSchemaForForm(
+  validationSchema: JSONSchema7 = {},
+  entitySchemaBaseProperties: JSONSchema7['properties'] = undefined,
+): JSONSchema7 {
+  return {
+    $schema: 'http://json-schema.org/draft-07/schema#',
+    // Spread the validation schema to ensure definitions are included
+    // The definitions get lost if we simply put the entire schema in `allOf`
+    ...validationSchema,
+    allOf: [
+      ...(validationSchema.allOf || []),
+      // Add in the entity properties
+      {
+        type: 'object',
+        properties: entitySchemaBaseProperties,
+      },
+    ],
+    // Always allow adding additional annotations (though it may violate the validationSchema)
+    additionalProperties: true,
+  }
+}
+
+/**
+ * Generates the UiSchema for the annotations editor.
+ * @param entitySchemaBaseProperties
+ */
+export function getUiSchemaForForm(
+  entitySchemaBaseProperties: JSONSchema7['properties'] = {},
+) {
+  // Each 'base' entity field (id, name, concreteType, etc.) should be hidden
+  const uiSchemaWithHiddenEntityProperties = Object.keys(
+    entitySchemaBaseProperties,
+  ).reduce((prev, curr) => ({ ...prev, [curr]: { 'ui:widget': 'hidden' } }), {})
```

Comment:
Now that the entity fields are part of the schema/formData, we have to hide them. Easy to do with the UiSchema.